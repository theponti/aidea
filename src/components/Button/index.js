import cx from "classnames";
import t from "prop-types";
import React from "react";
import styles from "./Button.module.scss";

function Button({ children, className, variant, ...props }) {
  let buttonStyle;

  switch (variant) {
    case "success":
      buttonStyle = styles.btnSuccess;
      break;
    case "danger":
      buttonStyle = styles.btnDanger;
      break;
    default:
      buttonStyle = "";
  }

  return (
    <button
      data-testid="button"
      className={cx(styles.container, className, buttonStyle)}
      {...props}
    >
      {children}
    </button>
  );
}

Button.propTypes = {
  children: t.node,
  className: t.string,
  variant: t.oneOf(["success", "danger"]),
};

export default Button;
