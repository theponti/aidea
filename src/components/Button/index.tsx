import React, { ButtonHTMLAttributes } from 'react'
import styles from './Button.module.scss'
import cx from 'classnames'

interface Button extends ButtonHTMLAttributes<HTMLButtonElement> {
  variant: string;
}

function Button ({ children, className, variant, ...props }: Button) {
  let buttonStyle

  switch (variant) {
    case 'success':
      buttonStyle = styles.btnSuccess
      break
    case 'danger':
      buttonStyle = styles.btnDanger
      break
    default:
      buttonStyle = ''
  }

  return (
    <button
      data-testid="button"
      className={cx(styles.container, className, buttonStyle)}
      {...props}
    >
      {children}
    </button>
  )
}

export default Button
