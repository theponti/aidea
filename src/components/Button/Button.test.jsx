import { render } from "@testing-library/react";
import { describe, expect, it } from "vitest";
import Button from ".";

describe("<Button/>", () => {
  it("should render component", () => {
    const { getByTestId } = render(<Button />);
    expect(getByTestId("button")).toMatchSnapshot();
  });

  it("should render a success button", () => {
    const { container } = render(<Button variant="success" />);
    expect(container).toMatchSnapshot();
  });

  it("should render a danger button", () => {
    const { container } = render(<Button variant="danger" />);
    expect(container).toMatchSnapshot();
  });
});
