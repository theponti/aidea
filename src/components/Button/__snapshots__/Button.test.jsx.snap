// Vitest Snapshot v1

exports[`<Button/> > should render a danger button 1`] = `
<div>
  <button
    class="_container_1pont_1 _btnDanger_1pont_16"
    data-testid="button"
  />
</div>
`;

exports[`<Button/> > should render a success button 1`] = `
<div>
  <button
    class="_container_1pont_1 _btnSuccess_1pont_21"
    data-testid="button"
  />
</div>
`;

exports[`<Button/> > should render component 1`] = `
<button
  class="_container_1pont_1"
  data-testid="button"
/>
`;
